### PR TITLE
WIP Add option to specify if you want to run quick or long test for subst…

### PR DESCRIPTION
**Pull request by radupopa2010**
Add the possibility to specify which kind of test to run for substrate. 
There are quick tests, like the ["smoke test"](https://gitlab.parity.io/parity/simnet/-/blob/master/testing/substrate_tests/tests/quick/001-smoketest.feature), or longer tests, like the ["load test"](https://gitlab.parity.io/parity/simnet/-/blob/master/testing/substrate_tests/tests/long/002-loadtest.feature)

For PRs we might want to run the quick tests, while for the nightly scheduled job the long test

### CODE REVIEW

File: .gitlab-ci.yml [hunk 1]
```diff
@@ -711,4 +711,15 @@ trigger-simnet:
...
+    # use the -v operator to explicitly test if a variable is set
+    # use the -n operator to test if variable is empty string
+    - if [[ -v FEATURES && -n "${FEATURES}" ]]  ; then 
+        echo "Testing only ${FEATURES}" ;
+        SUBSTRATE_FEATURES="--features=${FEATURES}"   ;
+      else  
+        echo "Value for FEATURES not specified, all tests will run" ;
+        SUBSTRATE_FEATURES="" ;
+      fi
+
```
Comment (TriplEight) [inline .gitlab-ci.yml:723]:
```suggestion
```

File: .maintain/gitlab/trigger_pipeline.sh [hunk 2]
```diff
@@ -88,7 +91,7 @@ function main {
 
 function parse_args {
   # shellcheck disable=SC2214
-  while getopts c:u:r:i:n:g:t:r:a:s:h-: OPT; do
+  while getopts c:u:r:i:n:g:f:t:r:a:s:h-: OPT; do
```
Comment (TriplEight) [inline .maintain/gitlab/trigger_pipeline.sh:94]:
```suggestion
  while getopts f:r:a:c:t:u:r:i:n:g:s:h-: OPT; do
```

**Review by TriplEight — CHANGES_REQUESTED** [review]
I think at this point it's already too much logic stored on the wrong side.
Besides, you have to maintain it in two repos, what if they will be more? And with this PR they will already be out of sync.
Most of this program has to reside within Simnet and CI should pass only variables depending on when it runs.

We've been discussing this before, we should stop triggering another CI from here and start passing instructions directly to Simnet.

File: .maintain/gitlab/trigger_pipeline.sh (hunk 2 above)
Comment (radupopa2010) [reply .maintain/gitlab/trigger_pipeline.sh:94]: you really like scrabble :))

**Comment by radupopa2010** [comment]
Yes, we need to schedule a meeting to speak how is better to implement the changes.

**Review by TriplEight — APPROVED** [review]

File: .gitlab-ci.yml [hunk 3]
```diff
@@ -711,4 +711,14 @@ trigger-simnet:
     DWNSTRM_ID:                    332
   script:
     # API trigger for a simnet job
-    - .maintain/gitlab/trigger_pipeline.sh --simnet-version=${SIMNET_REF}
+    # use the -v operator to explicitly test if a variable is set
+    # use the -n operator to test if variable is empty string
+    - if [[ -v FEATURES && -n "${FEATURES}" ]]  ; then
```
Comment (TriplEight) [inline .gitlab-ci.yml:716]: It's better to just have a conditional flag in the script than writing additional scripts around it

**Review by TriplEight — CHANGES_REQUESTED** [review]

**Pull request closed without merging**